demo-server/src/main/java/com/illia/server/file_holder/FileHolderImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.illia.server.file_holder;
+
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class FileHolderImpl implements FileHolder {
+
+    private final Map<String, File> savedFiles = new HashMap<>();
+
+    @Override
+    public File getFile(String fileName) {
+        return savedFiles.get(fileName);
+    }
+
+    @Override
+    public File saveFile(String fileName, File file) {
+        savedFiles.put(fileName, file);
```

Comment:
add some validation here; 
and try writing file on the filesystem to not keep whole file in-memory. It is usually done by saving on the cloud or DB, but let's practice file disk operations (read/write)
I think I remember you working with AWS, you can try saving it on S3 if you like